### ISSUE TITLE
Bug in Learn many for Linear Regression

### ISSUE THREAD

**Issue by philippkiesling** [online-ml/river#996]
## Versions

**river version**: 0.11.1
**Python version**: 3.9 
**Operating system**: Windows 10
**scipy**: 1.9.0
**numpy**: 1.22.4

We are getting very unusual results when using `learn_many` with linear regression. The results are very different from when training in a loop and very inaccurate. 

I was able to reproduce the bug with the Trump Example from river datasets: 

```
from river import datasets
from river import evaluate
from river import linear_model
from river import metrics
from river import preprocessing

dataset = datasets.TrumpApproval()
#create Pandas Dataset
l=[]
target = []
for X, y in dataset: 
    l.append([i for i in X.values()])
    target.append(y)
batch_data = pd.DataFrame(np.array(l), columns = X.keys())

#Learn model using learn_one only
model1 = (
    preprocessing.StandardScaler() |
    linear_model.LinearRegression()
)
for X,y in dataset: 
    model1.learn_one(X,y)
intercept = model1["LinearRegression"].intercept
weights = model1["LinearRegression"].weights
print(f"Intercept: {intercept} Weights{weights}")

#Learn model using learn_many only

model2 = (
    preprocessing.StandardScaler() |
    linear_model.LinearRegression()
)
model2.learn_many(batch_data, pd.Series(target))
model2.predict_many(batch_data)
intercept = model2["LinearRegression"].intercept
weights = model2["LinearRegression"].weights
print(f"Intercept: {intercept} Weights{weights}")


```

Model1: 
Intercept: 41.5687238707464 Weights{'ordinal_date': 0.0, 'gallup': 0.0, 'ipsos': 0.0, 'morning_consult': 0.0, 'rasmussen': 0.0, 'you_gov': 0.0}
Mpdel2: 
Intercept: 0.8150916109090911 Weights{'ordinal_date': 600637.2658032685, 'gallup': 33.30937762916284, 'ipsos': 33.23264291914282, 'morning_consult': 33.48822098365387, 'rasmussen': 33.27196516367656, 'you_gov': 33.2930695367527}

**Comment by kulbachcedric** [comment]
Hi Philipp,

It is noticeable that the weights of the incrementally learned model are 0.0. Have you tried passing an initializer e.g. `river.optim.initializer.Normal`?

@MaxHalford: We could have a default initializer for the LR model? 

However, I think the results may not necessarily be identical, as the order of the data points might change within the batch. 
@jacobmontiel had a great [talk](https://github.com/jacobmontiel/river-tutorial-ijcai2020/blob/master/demo-river.ipynb) on that.

Best
Cedric

**Comment by MaxHalford** [comment]
Hey hey. Apologies for the delayed answer.

This isn't actually a bug. How `learn_many` works:

1. `y_pred` is calculated for each `x` in the batch
2. the error between `y` and `y_pred` is calculated
3. the errors are averaged
4. the average error is used to build a gradient
5. the gradient is used to update the weights

With `learn_one`, this process is applied for each observation. If you try with PyTorch or with scikit-learn you'll arrive at the same result: the batch size doesn't just affect the speed, it also changes the model parameters. In other words, `n * batch(1) != 1 * batch(n)`.

Does that help? Feel free to ask follow-up with questions if it's not clear.
